libs/js-sdk/src/utils/compare-arrays.ts

```diff
@@ -0,0 +1,6 @@
+export function compareArrays<T>(a: T[], b: T[]) : boolean {
+    if (a.length !== b.length) {
+        return false
+    }
+    return a.filter((x): x is T => !!x).sort() === b.filter((y): y is T => !!y).sort()
```

Comment:
This will always return false because they will be different object references, maybe something like

```
const sorted1 = a.filter((x): x is T => !!x).sort()
const sorted2 = b.filter((y): y is T => !!y).sort()

sorted1.every((v, i) => v === sorted2[i])
```